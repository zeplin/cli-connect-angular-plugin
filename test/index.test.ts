import Plugin from "../src";

describe("Connected Components Angular Plugin", () => {
    beforeEach(() => {
        jest.resetModules();
    });

    test("component.ts snippet creation", async () => {
        const plugin = new Plugin();

        const componentCode = await plugin.process(
            {
                path: "test/samples/component.ts",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("componentWithDecoratorInputs.ts snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/componentWithDecoratorInputs.ts",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("componentWithFileTemplate.ts snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/componentWithFileTemplate.ts",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("componentWithMultiSelectors.ts snippet creation", async () => {
        const processor = new Plugin();

        const componentCode = await processor.process(
            {
                path: "test/samples/componentWithMultiSelectors.ts",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });

    test("componentWithMultiSelectors.ts snippet creation", async () => {
        const processor = new Plugin();
        processor.init({ config: { useFullSnippet: true, useFullDescription: true } });

        const componentCode = await processor.process(
            {
                path: "test/samples/componentWithMultiSelectors.ts",
                zeplinNames: []
            }
        );

        expect(componentCode).toMatchSnapshot();
    });
});
