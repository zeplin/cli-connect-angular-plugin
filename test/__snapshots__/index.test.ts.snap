// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Components Angular Plugin component.ts snippet creation 1`] = `
Object {
  "description": "#### BankAccount
This is a bank account component
This is line 2 description


#### BankAccountHolder
This is a bank account holder component",
  "lang": "markup",
  "snippet": "<!--- BankAccount examples --->
<!--- Possible selectors:
  - bank-account
--->
<bank-account
  account-id='{string}'
  bankName='{string}'>
</bank-account>



<!--- BankAccountHolder examples --->
<!--- Possible selectors:
  - bank-account-holder
--->
<bank-account-holder>
    {content}
</bank-account-holder>",
}
`;

exports[`Connected Components Angular Plugin componentWithDecoratorInputs.ts snippet creation 1`] = `
Object {
  "description": "#### BankAccount
This is a bank account component


#### BankAccountHolder
This is a bank account holder component",
  "lang": "markup",
  "snippet": "<!--- BankAccount examples --->
<!--- Possible selectors:
  - bank-account
--->
<bank-account
  account-id='{string}'
  bankName='{string}'>
</bank-account>



<!--- BankAccountHolder examples --->
<!--- Possible selectors:
  - bank-account-holder
--->
<bank-account-holder>
    {content}
</bank-account-holder>",
}
`;

exports[`Connected Components Angular Plugin componentWithFileTemplate.ts snippet creation 1`] = `
Object {
  "description": "#### BankAccount
This is a bank account component


#### BankAccountHolder
This is a bank account holder component",
  "lang": "markup",
  "snippet": "<!--- BankAccount examples --->
<!--- Possible selectors:
  - bank-account
--->
<bank-account
  account-id='{string}'
  bankName='{string}'>
</bank-account>



<!--- BankAccountHolder examples --->
<!--- Possible selectors:
  - bank-account-holder
--->
<bank-account-holder>
    {content}
</bank-account-holder>",
}
`;

exports[`Connected Components Angular Plugin componentWithMultiSelectors.ts snippet creation 1`] = `
Object {
  "description": "#### BankAccount
This is a bank account component


#### BankAccountHolder
This is a bank account holder component",
  "lang": "markup",
  "snippet": "<!--- BankAccount examples --->
<!--- Possible selectors:
  - bank-account
  - .account
  - [type=account]
  - p.statement[type=account]
--->
<bank-account
  account-id='{string}'
  bankName='{string}'>
</bank-account>



<!--- BankAccountHolder examples --->
<!--- Possible selectors:
  - bank-account-holder
--->
<bank-account-holder>
    {content}
</bank-account-holder>",
}
`;

exports[`Connected Components Angular Plugin componentWithMultiSelectors.ts snippet creation 2`] = `
Object {
  "description": "# BankAccount
This is a bank account component

# BankAccountHolder
This is a bank account holder component",
  "lang": "markup",
  "snippet": "<!--- BankAccount examples --->
<bank-account
  account-id='{string}'
  bankName='{string}'>
</bank-account>

<div
  class='account'
  account-id='{string}'
  bankName='{string}'>
</div>

<div
  type='account'
  account-id='{string}'
  bankName='{string}'>
</div>

<p
  type='account'
  class='statement'
  account-id='{string}'
  bankName='{string}'>
</p>



<!--- BankAccountHolder examples --->
<bank-account-holder>
    {content}
</bank-account-holder>",
}
`;
